cmd/egress/cmd.go

```diff
@@ -158,22 +93,101 @@ are set correctly before execution.
 				config.CaCert = bytes.NewBuffer(cert).String()
 			}
 
-			p := proxy.ProxyConfig{
+			proxy := proxy.ProxyConfig{
 				HttpProxy:  config.httpProxy,
 				HttpsProxy: config.httpsProxy,
 				Cacert:     config.CaCert,
 				NoTls:      config.noTls,
 			}
 
-			out := cli.ValidateEgress(ctx, config.vpcSubnetID, config.cloudImageID, config.kmsKeyID, config.securityGroupId, config.timeout, p)
+			// setup non cloud config options
+			vei := verifier.ValidateEgressInput{
+				Ctx:          context.TODO(),
+				SubnetID:     config.vpcSubnetID,
+				CloudImageID: config.cloudImageID,
+				Timeout:      config.timeout,
+				Tags:         config.cloudTags,
+				InstanceType: config.instanceType,
+				Proxy:        proxy,
+			}
+
+			// AWS workflow
+			if !config.gcp {
+
+				//Setup AWS Specific Configs
+				vei.AWS = verifier.AwsEgressConfig{
+					KmsKeyID:        config.kmsKeyID,
+					SecurityGroupId: config.securityGroupId,
+				}
+
+				awsVerifier, err := utils.GetAwsVerifier(config.region, config.awsProfile, config.debug)
+				if err != nil {
+					fmt.Printf("could not build awsVerifier %v", err)
+					os.Exit(1)
+				}
+
+				awsVerifier.Logger.Warn(context.TODO(), "Using region: %s", config.region)
+
+				out := verifier.ValidateEgress(awsVerifier, vei)
+				out.Summary(config.debug)
+
+				if !out.IsSuccessful() {
+					awsVerifier.Logger.Error(context.TODO(), "Failure!")
+					os.Exit(1)
+				}
+
+				awsVerifier.Logger.Info(context.TODO(), "Success")
+				os.Exit(0)
+			}
+
+			// check for empty env vars
+			// GCP workflow
+			if config.gcp {
+				projectID := os.Getenv("MONGO_PASS")
+				if projectID != "" {
+					fmt.Println("please set environment variable GCP_PROJECT_ID to the project ID of the VPC")
+					os.Exit(1)
+				}
+				vpcName := os.Getenv("MONGO_PASS")
```

Comment:
the same q as above, probably we'd need to reword to `GCP_VPC_NAME` and `GCP_PROJECT_ID`